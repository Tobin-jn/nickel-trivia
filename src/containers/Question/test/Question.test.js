import React from 'react'
import { shallow } from 'enzyme'

describe('Question', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Question />)
  })
  
  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})