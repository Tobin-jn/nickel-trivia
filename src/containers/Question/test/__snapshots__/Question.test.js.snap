// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    category="geogrpahy"
    currentQuestion={
      Object {
        "answers": Array [],
        "asked": false,
        "category": "Geography",
        "correct_answer": "Washington D.C",
        "id": 1,
        "question": "Which city is the capital of the United States of America?",
      }
    }
    questions={
      Array [
        Object {
          "answers": Array [],
          "asked": false,
          "category": "Geography",
          "correct_answer": "Washington D.C",
          "id": 1,
          "question": "Which city is the capital of the United States of America?",
        },
        Object {
          "answers": Array [],
          "asked": false,
          "category": "Geography",
          "correct_answer": "Phoenix",
          "id": 2,
          "question": "What is the capital of the American state of Arizona?",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="question-container"
        >
          <div
            className="question-container-after"
          />
          <h3
            className="question"
          >
            Which city is the capital of the United States of America?
             
          </h3>
        </div>,
        <div
          className="answer-container"
        >
          <div
            onClick={[Function]}
          >
            <p
              className="answer-choice choice-a"
            />
          </div>
          <div
            onClick={[Function]}
          >
            <p
              className="answer-choice choice-b"
            />
          </div>
          <div
            onClick={[Function]}
          >
            <p
              className="answer-choice choice-c"
            />
          </div>
          <div
            onClick={[Function]}
          >
            <p
              className="answer-choice choice-d"
            />
          </div>
        </div>,
        <div
          className="answer-response"
        >
          <p>
            
          </p>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="question-container-after"
            />,
            <h3
              className="question"
            >
              Which city is the capital of the United States of America?
               
            </h3>,
          ],
          "className": "question-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "question-container-after",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Which city is the capital of the United States of America?",
                " ",
              ],
              "className": "question",
            },
            "ref": null,
            "rendered": Array [
              "Which city is the capital of the United States of America?",
              " ",
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-a"
              />
            </div>,
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-b"
              />
            </div>,
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-c"
              />
            </div>,
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-d"
              />
            </div>,
          ],
          "className": "answer-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="answer-choice choice-a"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "answer-choice choice-a",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="answer-choice choice-b"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "answer-choice choice-b",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="answer-choice choice-c"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "answer-choice choice-c",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="answer-choice choice-d"
              />,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "answer-choice choice-d",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            
          </p>,
          "className": "answer-response",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="question-container"
          >
            <div
              className="question-container-after"
            />
            <h3
              className="question"
            >
              Which city is the capital of the United States of America?
               
            </h3>
          </div>,
          <div
            className="answer-container"
          >
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-a"
              />
            </div>
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-b"
              />
            </div>
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-c"
              />
            </div>
            <div
              onClick={[Function]}
            >
              <p
                className="answer-choice choice-d"
              />
            </div>
          </div>,
          <div
            className="answer-response"
          >
            <p>
              
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="question-container-after"
              />,
              <h3
                className="question"
              >
                Which city is the capital of the United States of America?
                 
              </h3>,
            ],
            "className": "question-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "question-container-after",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Which city is the capital of the United States of America?",
                  " ",
                ],
                "className": "question",
              },
              "ref": null,
              "rendered": Array [
                "Which city is the capital of the United States of America?",
                " ",
              ],
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                onClick={[Function]}
              >
                <p
                  className="answer-choice choice-a"
                />
              </div>,
              <div
                onClick={[Function]}
              >
                <p
                  className="answer-choice choice-b"
                />
              </div>,
              <div
                onClick={[Function]}
              >
                <p
                  className="answer-choice choice-c"
                />
              </div>,
              <div
                onClick={[Function]}
              >
                <p
                  className="answer-choice choice-d"
                />
              </div>,
            ],
            "className": "answer-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="answer-choice choice-a"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "answer-choice choice-a",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="answer-choice choice-b"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "answer-choice choice-b",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="answer-choice choice-c"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "answer-choice choice-c",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="answer-choice choice-d"
                />,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "answer-choice choice-d",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              
            </p>,
            "className": "answer-response",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
